include/oneapi/dpl/pstl/algorithm_impl.h

```diff
@@ -1487,10 +1492,13 @@ __remove_elements(_ExecutionPolicy&& __exec, _ForwardIterator __first, _ForwardI
             ::std::forward<_ExecutionPolicy>(__exec), __result, __result + __m,
             [__result, __first, __is_vector](_Tp* __i, _Tp* __j) {
                 __invoke_if_else(
-                    ::std::is_trivial<_Tp>(),
+                    ::std::is_trivially_destructible<_Tp>(),
```

Comment:
You didn't optimize it, because parallel loop call is still there

Reply:
The goal of change at this line is that more correct to use is_trivially_destructible for solution about whould we call __brick_move or __brick_move_destroy.
You can see this in example at https://godbolt.org/z/r3YGsKME8

Reply:
But parallel loop call required here in any case: we should run __brick_move or __brick_move_destroy